lib/ecto/repo.ex

```diff
@@ -498,6 +506,39 @@ defmodule Ecto.Repo do
               {:ok, Ecto.Model.t} | {:error, Ecto.Changeset.t}
 
   @doc """
+  Inserts or updates a model or changeset depending on whether the model
+  is persisted or not.
+
+  The distinction whether to insert or update will be made on the
+  `Ecto.Schema.Metadata` field `:state`. The `:state` is automatically set by
+  Ecto when loading or building a schema.
+
+  Please note that for this to work, you will have to load existing models from
+  the database. So even if the model exists, this won't work:
+
+      model = %Post{id: 'existing_id', ...}
+      MyRepo.insert_or_update model
+      # => {:error, "id already exists"}
+
+  ## Example
+
+      model =
+        case MyRepo.get(Post, id) do
+          nil  -> %Post{id: id} # Post not found, we create a new one
```

Comment:
Super nitpicky, but maybe `we *build* one` instead. I think `built` is typically the Ecto terminology for unsaved records.